testing/conformance/tests/runner.rs

```diff
@@ -74,76 +75,64 @@ async fn conformance_test_runner() -> anyhow::Result<()> {
-                    .with_context(|| format!("failed to run vector: {}", path.display()))?;
-                anyhow::Ok((path, res))
-            })
-            .map(futures::future::Either::Left),
-        )
+    // Collect test vector files
+    let vector_paths = if path.is_file() {
+        vec![path]
     } else {
-        either::Either::Right(
-            WalkDir::new(path)
-                .into_iter()
-                .filter_ok(is_runnable)
-                .map(|e| {
-                    let stats = stats.clone();
-                    let tracer = tracer.clone();
-                    async move {
-                        let path = e?.path().to_path_buf();
-                        let res = run_vector(path.clone(), stats, tracer)
-                            .await
-                            .with_context(|| format!("failed to run vector: {}", path.display()))?;
-                        Ok((path, res))
-                    }
-                })
-                .map(futures::future::Either::Right),
-        )
+        WalkDir::new(path)
+            .into_iter()
+            .filter_ok(is_runnable)
+            .map(|e| e.map(|de| de.path().to_path_buf()))
+            .collect::<Result<Vec<_>, _>>()?
     };
 
-    let mut results = Box::pin(
-        stream::from_iter(vector_results)
-            // Will _load_ up to 100 vectors at once in any order. We won't actually run the vectors in
-            // parallel (yet), but that shouldn't be too hard.
-            .map(|task| {
-                async move {
-                    let (path, jobs) = task.await?;
-                    Ok(stream::from_iter(jobs).map(move |job| {
-                        let path = path.clone();
-                        Ok(async move { anyhow::Ok((path, job.await?)) })
-                    }))
-                }
-                .try_flatten_stream()
-            })
-            .flatten()
-            .try_buffer_unordered(*TEST_VECTOR_PARALLELISM),
-    );
+    let stats = Arc::new(stats);
+    let tracer = Arc::new(tracer);
+
+    // Process all test vectors concurrently, limited by buffer_unordered
+    let mut results = stream::iter(vector_paths)
+        .map(|path| {
+            let stats = stats.clone();
+            let tracer = tracer.clone();
+
+            async move {
+                // Run the vector processing in a blocking task
+                tokio::task::spawn_blocking(move || run_vector(path, stats, tracer)).await?
+            }
+        })
+        .buffer_unordered(*TEST_VECTOR_PARALLELISM)
+        .collect::<Vec<_>>()
+        .await;
 
     let mut succeeded = 0;
     let mut failed = 0;
     let mut skipped = 0;
 
-    while let Some((path, res)) = results.next().await.transpose()? {
-        match res {
-            VariantResult::Ok { id } => {
-                report!("OK".on_green(), path.display(), id);
-                succeeded += 1;
+    // Process results
+    for result in results.drain(..) {
```

Comment:
Any reason for a `drain` here and not an `into_iter` or something like this? Do we re-use the `results` anywhere later?

Reply:
nope, good point! changed